### PR TITLE
Unification tests (not passing yet)

### PR DESCRIPTION
**Pull request by dfilaretti**
I'm trying to add some simple tests for unification of `set`, `list` and `map`. 
Although there is a different test for each, the logic is the same. 
They don't work yet, and they all seem to fail for the same reason. 

```
kore-exec: Unexpected non-false predicate GenericPredicate (Fix (AndPattern (And {andSort = SortActualSort (SortActual ...
```

(needs more investigation)

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: src/main/k/in-progress/tests/list-unify-1/tests/2.output.golden [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<generatedTop>
```
Comment (traiansf) [inline src/main/k/in-progress/tests/list-unify-1/tests/2.output.golden]: I think the output here would include b, as (unlike for set) you're only removing from one end.

File: src/main/k/in-progress/tests/list-unify-1/tests/3.output.golden [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<generatedTop>
```
Comment (traiansf) [inline src/main/k/in-progress/tests/list-unify-1/tests/3.output.golden]: Same comment as above

**Comment by dfilaretti** [comment]
I also realized I forgot to test the simple case of two concrete sets/lists. E.g. unifying a set with itself. I'm going to add those as well.

File: src/main/k/in-progress/tests/list-unify-1/tests/2.output.golden (hunk 1 above)
Comment (dfilaretti) [reply src/main/k/in-progress/tests/list-unify-1/tests/2.output.golden]: you are right, I got confused between lists and sets. fixed now.

**Pull request merged**